src/parameters.jl

```diff
@@ -14,13 +14,71 @@ An abstract type that represents a set of multiple parameters.
 
 abstract type AbstractParameterSet end
 
+"""Returns the number of parameters in a parameter set."""
+function length(::T) where {T <: AbstractParameterSet}
+  return length(fieldnames(T))
+end
+
+"""Set names of parameters to their fieldnames."""
+function set_names!(parameter_set::T) where {T <: AbstractParameterSet}
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    p.name = string(param_name)
+  end
+end
+
+"""Returns current values of each parameter in a parameter set."""
+function values(parameter_set::T) where {T <: AbstractParameterSet}
+  values = []
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    push!(values, value(p))
+  end
+  return values
+end
+
+"""Returns lower bounds of each parameter in a parameter set."""
+function lower_bounds(parameter_set::T) where {T <: AbstractParameterSet}
+  lower_bounds = []
+  return lower_bounds!(parameter_set, lower_bounds)
+end
+
+"""Returns lower bounds of each parameter in a parameter set in place."""
+function lower_bounds!(parameter_set::T, vals::AbstractVector) where {T <: AbstractParameterSet}
+  empty!(vals)
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    d = domain(p)
+    push!(vals, lower(d))
+  end
+  return vals
+end
```

Comment:
```suggestion
function lower_bounds(parameter_set::T) where {T <: AbstractParameterSet}
  lower_bounds = Vector{Any}(undef, length(parameter_set))
  return lower_bounds!(parameter_set, lower_bounds)
end

"""Returns lower bounds of each parameter in a parameter set in place."""
function lower_bounds!(parameter_set::T, vals::AbstractVector) where {T <: AbstractParameterSet}
  # check that length(vals) and length(parameter_set) are the same
  empty!(vals) # not needed
  for (i, param_name) in zip(1:length(vals), fieldnames(T))
    p = getfield(parameter_set, param_name)
    d = domain(p)
    vals[i] = lower(d)
  end
  return vals
end
```

Reply:
The idea is more to have vector of fixed length whenever we can.
The best would be to be more specific with `Vector{Any}` in some cases.